src/protocols/kubernetes/url.c

```diff
@@ -89,15 +89,33 @@ int guac_kubernetes_escape_url_component(char* output, int length,
 
 }
 
-int guac_kubernetes_endpoint_attach(char* buffer, int length,
+int guac_kubernetes_append_endpoint_param(char* buffer, int length, 
+        const char* param_name, const char* param_value) {
+
+    char escaped_param_value[GUAC_KUBERNETES_MAX_ENDPOINT_LENGTH];
+
+    /* Escape value */
+    if (guac_kubernetes_escape_url_component(escaped_param_value,
+                    sizeof(escaped_param_value), param_value))
+            return 1;
+    
+    int written;
+    written = snprintf(buffer + strlen(buffer), length - strlen(buffer), 
+            "%s=%s&", param_name, escaped_param_value);
```

Comment:
Rather than iterate through the string twice (by calling `strlen()` twice), and rather than assuming that further parameters will be appended and always including an "&", I suggest killing two birds with one stone and determining the length while determining whether the "?" is already present (and thus whether a "?" or "&" prefix is needed).

This function would then be usable in any context, perhaps appending directly to the path being built rather than the awkward:

```c
    written = snprintf(buffer, length, "%s?%sstdin=true&stdout=true&tty=true", 
        endpoint_path, endpoint_params);
```

Reply:
Beware also that calculating the length passed to `snprintf()` as `length - strlen(buffer)` has the potential to write into memory off the end of the buffer:

Without `strlen()` being bounded by the length of the buffer, it may well return a value larger than `length` if `buffer` is not null-terminated. That would result in a negative value that is ultimately interpreted as a very large positive value when converted into the `size_t` used by `snprintf()`.

Reply:
7a1ba51baef5a6e8e8d85f6a5b68c022d6ba0fa1 I have added algorithm which determine the delimiter and calculate the buffer string length also added verification the buffer null terminated, not sure 100% it is correct.
7809447c3ffd7c0fddef807231e994f52a6c15ed Added usage of modified function.